nomad/worker.go

```diff
@@ -215,9 +215,20 @@ func (w *Worker) waitForIndex(index uint64, timeout time.Duration) error {
 	start := time.Now()
 	defer metrics.MeasureSince([]string{"nomad", "worker", "wait_for_index"}, start)
 CHECK:
+	// Snapshot the current state
+	snap, err := w.srv.fsm.State().Snapshot()
```

Comment:
For this use case would it be worth being able to query this without a snapshot since you don't otherwise use the snapshot here?